lib/ansible/modules/network/nxos/nxos_vrf.py

```diff
@@ -242,21 +242,21 @@ def map_obj_to_commands(updates, module):
         elif state == 'present':
             if not obj_in_have:
                 commands.append('vrf context {0}'.format(name))
-                if rd and rd != '':
-                    commands.append('rd {0}'.format(rd))
-                if description:
-                    commands.append('description {0}'.format(description))
-                if vni and vni != '':
-                    commands.append('vni {0}'.format(vni))
+                for item in args:
+                    candidate = w.get(item)
+                    if candidate:
+                        cmd = item + ' ' + str(candidate)
+                        commands.append(cmd)
                 if admin_state == 'up':
                     commands.append('no shutdown')
                 elif admin_state == 'down':
                     commands.append('shutdown')
 
                 if commands:
                     if vni:
-                        if have.get('vni') and have.get('vni') != '':
-                            commands.insert(1, 'no vni {0}'.format(have['vni']))
+                        for h in have:
+                            if h.get('vni'):
+                                commands.insert(1, 'no vni {0}'.format(h['vni']))
```

Comment:
Can you please add comments why we need to unconfigure vni is some another vrf which is already configured on box ?

Reply:
+1